test/integration.sh

```diff
@@ -1,47 +1,89 @@
+    #       the required number of containers
+    #   (ii) not all exact names of required Docker containers
+    #        match all names in the list of running Docker containers
+
+    # Get array of running containers.
+    containers=($(sudo docker ps | tail -n +2 | awk ' {print $NF} '))
+
+    # Get array of all required containers from file.
+    fname=expected_containers
+    IFS=$'\r\n' GLOBIGNORE='*'\
+       command eval  'containers_expected=($(cat $fname))'
+    containers_expected=($containers_expected)  # cast as array
 
-    [[ -d outputs ]] && sudo rm -rf outputs && INFO "deleted old outputs"
+    num_containers_exp="${#containers_expected[@]}"
 
-    access_token=$(../redwood/cli/bin/redwood token create)
-    INFO "generated testing access token: ${access_token}"
+    # Crude initial test to verify that at least the minimum number of
+    # containers are running.
+    [ "${#containers[@]}" -ge $num_containers_exp ] ||\
+        (ERROR "Not running the required number of docker containers."\
+		&& exit 1)
 
-    redwood_endpoint=$(cat ../redwood/.env | grep 'base_url=' | sed 's/[^=]*=//')
-    INFO "got redwood endpoint from dcc-ops/redwood/.env: ${redwood_endpoint}"
+    for container_expected in "${containers_expected[@]}"; do
+        counter=0
+        for container in "${containers[@]}"; do
+            if [[ $container_expected = $container ]]; then
+                continue
+            else
+                counter=$((counter+1))
+            fi
+        done
 
-    INFO "doing upload with $(pwd)/manifest.tsv"
-    sudo docker run --rm -it -e ACCESS_TOKEN=${access_token} -e REDWOOD_ENDPOINT=${redwood_endpoint} \
-         -v $(pwd)/manifest.tsv:/dcc/manifest.tsv -v $(pwd)/samples:/samples -v $(pwd)/outputs:/outputs \
-         quay.io/ucsc_cgl/core-client:1.1.0-alpha spinnaker-upload --skip-submit --force-upload /dcc/manifest.tsv
+        if [[ "$counter" -eq $num_containers_exp ]]; then
+            ERROR "One or more docker containers not running."\
+		&& exit 1
+        fi
+    done
 
-    object_id=$(cat outputs/receipt.tsv | tail -n +3 | head -n 1 | cut -f 20)
-    bundle_id=$(cat outputs/receipt.tsv | tail -n +3 | head -n 1 | cut -f 19)
-    INFO "the uploaded bundle with id ${bundle_id} has metadata.json with object id ${object_id}"
+}
+
+function main {
+    # This tests whether all Docker containers are installed and then
+    # hits the DCC Dashboard Host and checks whether it returns a 200
+    # status code.
+
+    cd "$(dirname "${BASH_SOURCE[0]}")"
+    LOG_LEVEL_ALL
+    check_setup
 
-    INFO "downloading metadata.json with object id ${object_id}"
-    sudo mkdir outputs/test_download
-    sudo docker run --rm -it -e ACCESS_TOKEN=${access_token} -e REDWOOD_ENDPOINT=${redwood_endpoint} \
-         -v $(pwd)/outputs:/outputs quay.io/ucsc_cgl/core-client:1.1.0-alpha download ${object_id} /outputs/test_download
+    # Get name of the DCC Dashboard Host.
+    curr_dir=$(pwd)
+    cd ../boardwalk
```

Comment:
This could just be `cd boardwalk`

Reply:
I'm not sure. The script `cd`s to the test directory (`cd "$(dirname "${BASH_SOURCE[0]}")"` is the first command in the script), which is on the same level as directory `boardwalk`. It returns 
```
ubuntu@ip-172-31-24-164:~/cgp-deployment/test⟫ cd boardwalk
bash: cd: boardwalk: No such file or directory
```

Reply:
Sorry, I was wrong on this.